nav2_recoveries/plugins/spin.cpp

```diff
@@ -119,10 +119,15 @@ Status Spin::onCycleUpdate()
   action_server_->publish_feedback(feedback_);
 
   double remaining_yaw = abs(cmd_yaw_) - abs(relative_yaw_);
-  if (remaining_yaw <= 0) {
+  if (remaining_yaw < 1e-6) {
```

Comment:
Why not 0?

Reply:
this was one of the tweaks I had to make to get it work. I think it is a numerical issue that propagates from our conversion from roll pitch yaw to quaternion when faking the footprint / tf / costmap.

Reply:
If I set this to 0 again, the logger (below, lines 125-128) will just say `Remaining Yaw: 0.000000` and the recovery gets stuck there. I figured `1e-6` was acceptably close to zero to be OK. This is a quirk caused by the way I am faking the TF (I am not taking into account the `cmd_yaw` produced by the recoveries server).

If you disagree, can I suggest exposing this as a parameter?

Reply:
This is fine, just making sure there's a reasonable reason.

Reply:
That logging will need to be removed at some point before merging